dtfs-central-api/src/v1/controllers/utilisation-report-service/post-validate-utilisation-report-data.controller/index.ts

```diff
@@ -0,0 +1,32 @@
+import { ApiError, CustomExpressRequest } from '@ukef/dtfs2-common';
+import { Response } from 'express';
+import { HttpStatusCode } from 'axios';
+import { validateUtilisationReportCsvData } from '../../../../services/utilisation-report-data-validator';
+import { PostValidateUtilisationReportDataPayload } from '../../../routes/middleware/payload-validation/validate-post-validate-utilisation-report-data-payload';
+
+export type PostValidateUtilisationReportDataRequest = CustomExpressRequest<{
+  reqBody: PostValidateUtilisationReportDataPayload;
+}>;
+
+/**
+ * Controller to handle post request to validate utilisation report data route
+ * @param req - The request
+ * @param res - The response
+ * @returns - The response with the body set to contain any error messages to display
```

Comment:
So this is something that may become relevant at some point - the return type of `RequestHandler` in `express` should be `void` or `Promise<void>` (as of express 5 I think). This is not a problem in this file, but can become a problem in routes files ie. `router.post(postValidateUtilisationReportData)` would display a typescript error. I know this is something that we have done in most of our controller files which would need to be fixed, but could we update this controller so that the return type is `void`

Reply:
I think I have resolved this by just removing the returns, let me know if that was not the correct solution!

Reply:
Ah ok that did not work because it sends a response but continues the current function. So the option is to move to having an empty return immediately following the call, but I'm not sure the internet agrees that is the nicest pattern and I do think perhaps this is standard e.g. https://dev.to/adamkatora/why-you-should-always-use-return-before-ressend-in-express-apis-and-applications-k9k#:~:text=In%20Express%2C%20calling%20the%20res,not%20exit%20the%20current%20function.

Reply:
That's interesting - it seems this is a semantic thing with `RequestHandler` as depending on where you use the type it has different behaviour (which means that returning the `res.send` call is actually fine so long as you don't say that the function itself is a `RequestHandler` type and use it directly in the `router.use` params). Agree that leaving this as is would be better for consistency either way